x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx

```diff
@@ -565,7 +565,7 @@ const CreateRulePageComponent: React.FC = () => {
             shouldLoadQueryDynamically={defineStepData.shouldLoadQueryDynamically}
             queryBarTitle={defineStepData.queryBar.title}
             queryBarSavedId={defineStepData.queryBar.saved_id}
-            thresholdFields={defineStepData.threshold.field}
+            thresholdFields={defineStepData?.threshold?.field || []}
```

Comment:
It's possible to remove that prop by reading `thresholdFields` via `useFormData` hook.